Egress task corrections and clarifications (#3315)
istio/istio.io: content/docs/tasks/traffic-management/egress/index.md

This task describes how to configure Istio to expose external services to Istio-enabled clients.

> You'll learn how to enable access to external services by defining

Reviewer: To make the goals of the task clearer, it would be better to use a list:
```suggestion
This task shows three ways to enable access to external services:
```

> [`ServiceEntry`](/docs/reference/config/istio.networking.v1alpha3/#ServiceEntry) configurations,

Reviewer: ```suggestion
- Configure a [service entry](/docs/reference/config/istio.networking.v1alpha3/#ServiceEntry).
```

> or alternatively, by bypassing the Istio proxy for a specific range of IPs or configuring it

Reviewer: ```suggestion
- Bypass the Envoy proxy for a specific range of IPs.
```

> to pass requests through to external services on ports that are not configured inside the mesh.

Reviewer: ```suggestion
- Configure the Envoy proxy to pass requests through to external services on ports that are not configured inside the mesh.
```

## Install Istio with access to all external services by default

An alternative to calling external services directly is to instruct the Istio proxy to pass through, instead
of block, calls to external services by default. This option allows you to start evaluating Istio quickly,
without controlling access to external
services, and decide to [configure access to external services](#controlled-access-to-external-services) later.

> Istio has an [installation option](/docs/reference/config/installation-options/)

Reviewer 2: I think it was a comment of @rcaballeromx to swap the order in the sentence.

Reviewer: It was, because the fact that the option exists is not as important as its purpose. What do you want to emphasize, the link to all the installation options or why should the reader look for the option?

In this task you looked at three ways to call external services from an Istio mesh:

> 1. Using a `ServiceEntry` to register an accessible external service inside the mesh. This is the

Reviewer: Don't use `ServiceEntry` to refer to all service entries. It is best to use it when referring to a specific instance in a yaml file. Opt instead for the normal service entry. 

```suggestion
1. Use a service entry to register an accessible external service inside the mesh. This is the
```

   recommended approach.

1. Configuring the Istio sidecar to exclude external IPs from its remapped IP table.

1. Configuring Istio to allow access to any external service on some ports.

The first approach lets you use all of the same Istio service mesh features for calls to services inside or outside of

> the cluster. You saw how one can monitor access to external services and set a timeout rule for calls to an external

Reviewer: ```suggestion
the cluster. You saw how to monitor access to external services and set a timeout rule for calls to an external
```

service.

The second approach bypasses the Istio sidecar proxy, giving your services direct access to any external server.
However, configuring the proxy this way does require cluster-provider specific knowledge and configuration.
Additionally, you lose monitoring of access to external services and you can't apply Istio features on traffic to
external services.

The third approach directs traffic through the Istio sidecar proxy but it allows access to any service on any
port that has no HTTP service or service entry defined in the mesh. Similar to the second approach,
you can't monitor access to external services but you don't need to
know which IP ranges are external to the cluster.

> You can also easily switch to the first approach

Reviewer: ```suggestion
To easily switch to the first approach for a specific port,
```

> for a specific port by simply creating a service entry for that port.

Reviewer: ```suggestion
simply create a service entry for that port.
```

> This way you can use this approach initially, allowing access to any external service,

Reviewer: ```suggestion
This means, you can use this approach initially and allow access to any external service.
```

> and then later decide to start controlling access to external services for specific ports.

Reviewer: ```suggestion
Later, you can decide to start controlling access to external services for specific ports and
```

> You can enable traffic monitoring and control features as they are needed.

Reviewer: ```suggestion
enable traffic monitoring and control features as they are needed.
```
